search-service/src/test/kotlin/com/egm/stellio/search/config/TestContainersConfiguration.kt

```diff
@@ -0,0 +1,45 @@
+package com.egm.stellio.search.config
+
+import com.egm.stellio.shared.config.TestContainers
+import io.r2dbc.spi.ConnectionFactories
+import io.r2dbc.spi.ConnectionFactory
+import io.r2dbc.spi.ConnectionFactoryOptions
+import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.context.annotation.Bean
+
+@TestConfiguration
+class TestContainersConfiguration {
+
+    private val DB_NAME = "stellio_search"
+    private val DB_USER = "stellio_search"
+    private val DB_PASSWORD = "stellio_search_db_password"
+
+    object SearchServiceTestContainers : TestContainers("postgres", 5432) {
```

Comment:
Maybe a PostgresConfiguration could be in shared with 3 methods : 
getDBName, getDBUser, getDBPassword to define 
```java
abstract class TestContainersConfiguration {

    abstract fun getDBName()
    abstract fun getDBUser()
    abstract fun getDBPassword()

    object PostgresTestContainers : TestContainers("postgres", 5432) {
Attach files by dragging & dropping, selecting or pasting them.

        fun getPostgresqlHost(): String {
            return instance.getServiceHost(serviceName, servicePort)
        }

        fun getPostgresqlPort(): Int {
            return instance.getServicePort(serviceName, servicePort)
        }

        fun getPostgresqlUri(): String {
            return "jdbc:postgresql://" + getPostgresqlHost() + ":" + getPostgresqlPort() + '/'
        }
    }

    @Bean
    fun configuration(): ConnectionFactory {
        val options = ConnectionFactoryOptions.builder()
            .option(ConnectionFactoryOptions.DATABASE, getDBName())
            .option(ConnectionFactoryOptions.HOST, SearchServiceTestContainers.getPostgresqlHost())
            .option(ConnectionFactoryOptions.PORT, SearchServiceTestContainers.getPostgresqlPort())
            .option(ConnectionFactoryOptions.USER, getDBUser())
            .option(ConnectionFactoryOptions.PASSWORD, getDBPassword())
            .option(ConnectionFactoryOptions.DRIVER, "postgresql")
            .build()

        return ConnectionFactories.get(options)
    }
}
```

Reply:
Maybe this can avoid some code duplication but i think it's better to have separate configurations per service.
Open question